tests/probes/yamlfilecontent/test_probes_yamlfilecontent_key.sh

```diff
@@ -0,0 +1,33 @@
+#!/usr/bin/env bash
+
+. $builddir/tests/test_common.sh
+
+function test_probes_yamlfilecontent_key {
+
+    probecheck "yamlfilecontent" || return 255
+
+    local ret_val=0;
+    local DF="${srcdir}/test_probes_yamlfilecontent_key.xml"
+    local RF="results.xml"
+
+    [ -f $RF ] && rm -f $RF
+
+    cp "${srcdir}/openshift-logging.yaml" /tmp
+
+    local YAML_FILE="/tmp/openshift-logging.yaml"
+
+    $OSCAP oval eval --results $RF $DF
+
+    if [ -f $RF ]; then
+	verify_results "def" $DF $RF 2 && verify_results "tst" $DF $RF 3
+	ret_val=$?
+    else
+	ret_val=1
+    fi
+
+    rm -f $YAML_FILE
+
+    return $ret_val
```

Comment:
Where do you test that the correct data was extracted from the YAML? Please add assertions on collected items.

Reply:
The tests have `states`, if they pass doesn't it mean we have collected the correct data?

Reply:
Probably yes

Reply:
What about the that expected error result? And what about that non-existing scalar? The test could assume that no objects are collected.

Reply:
Per `yamlpath` description in https://github.com/OVAL-Community/OVAL/pull/90/files#diff-ae56180beddebc6013e4387e9ed9fe0eR2118, `error` is the expected result when the path leads to a non-scalar.

The xpath field in [xmlfilecontent_object](https://github.com/OVALProject/Language/blob/master/docs/independent-definitions-schema.md#-xmlfilecontent_object-) behaves the same way.

Reply:
Oh, maybe @matejak is asking about how the test verifies that the result was `error`?
In [verify_results()](https://github.com/OpenSCAP/openscap/blob/maint-1.3/tests/test_common.sh.in#L170) the result of evaluation is compared with the results the comments.

Reply:
Thanks, I was not able to understand just by looking at [verify_results documentation](https://github.com/OpenSCAP/openscap/blob/maint-1.3/docs/contribute/testing.adoc#verify_results-function). That doc needs a refresh.